### ISSUE TITLE
Sites that don't work properly

### ISSUE THREAD

**Issue by spite** [spite/ShaderEditorExtension#4]
http://david.li/flow - doesn't even load the UI, even though it's calling createProgram
http://xnajs.com/examples/02/ - does load the UI, but doesn't apply changes. Probably it has to do with how the same source code is used in both shaders

**Comment by spite** [comment]
About http://xnajs.com/examples/02/

it turns out that a shader can't end in #endif (?). Adding a line break solves the issue (??).

**Comment by spite** [comment]
Updated on v.1.0.4

**Comment by coderitual** [comment]
I'm not sure if it's proper place but I've found 2 more scenarios (v 1.0.4 was used): 

Ad example 2 (http://xnajs.com/examples/02/):
![example 2](https://dl.dropboxusercontent.com/u/11187529/cr/issues/shader-editor-extension/issue-2.png)
After clicking on programs on the left pane, changing and undoing (ctr-z) one of my shader program stopped working. Above you can see that vertex shader differs from fragment one (the only difference should be define placed at the top). 

Ad example 1 (http://xnajs.com/examples/01/):
![example 2](https://dl.dropboxusercontent.com/u/11187529/cr/issues/shader-editor-extension/issue.png)
This issue is quite simple to reproduce. I have opened shader editor (page reloaded) and when I clicked on program 1 everything goes black. 

In both cases (after it becomes broken), when I clicked on the programs  I can see that another linebreak was added

**Comment by spite** [comment]
OK, that's good. If it's reproducible it can be fixed. I'll take at stab at it later. Thanks!

**Comment by ceane** [comment]
It does not work on this site for some reason: http://webgl-examples.appspot.com

**Comment by spite** [comment]
@Ceane you're right, it's not injecting the editor for some reason. But i see that my local version does inject successfully, there must be something getting in the way. Thanks for reporting!

**Comment by spite** [comment]
~~http://homepage.ntlworld.com/keir.clarke/web/webGLglobe.htm textures get messed up when selecting different programs~~

**Comment by spite** [comment]
https://sketchfab.com/models/ea4dcf0df7c444ea96c18684b4c380df shows the programs, but there's no source. Firefox doesn't even detect the created context.

**Comment by spite** [comment]
http://mountain.quechua.com/lookbook-spring-summer-15 - shaders break when optimising, but not when hightlighting

**Comment by spite** [comment]
https://my.sketchup.com/viewer/3dw?WarehouseModelId=e66bbf9037ec6e6d9971415b0eb153a&failureURL=aHR0cHM6Ly8zZHdhcmVob3VzZS5za2V0Y2h1cC5jb20vZW1iZWQuaHRtbD9taWQ9ZTY2YmJmOTAzN2VjNmU2ZDk5NzE0MTViMGViMTUzYSZldHA9c3cmcmVkaXJlY3Q9MQ__

**Comment by lcb931023** [comment]
https://www.shadertoy.com hangs on "waiting for programs to be added..."
